src/ewmh.cpp

```diff
@@ -460,10 +460,10 @@ bool Ewmh::isFullscreenSet(Window win) {
 }
 
 void Ewmh::setWindowOpacity(Window win, double opacity) {
-    uint32_t int_opacity = std::numeric_limits<uint32_t>::max()
+    long long_opacity = std::numeric_limits<long>::max()
                             * CLAMP(opacity, 0, 1);
 
-    X_.setPropertyCardinal(win, g_netatom[NetWmWindowOpacity], { int_opacity });
```

Comment:
I think this is wrong because the opacity value in the window property should not depend on the size of `long`. But I can't find the documentation for `_NET_WM_WINDOW_OPACITY`.

Reply:
The [`compton-trans`](https://github.com/chjj/compton/blob/master/bin/compton-trans#L206) utility sets the value to
```
opacity=$((opacity * 0xffffffff / 100))
```
So maybe we can simply multiply by `0xffffffff` instead of the limit of whatever type.

Reply:
Thanks, changed it and added a comment. Is this correct now?

Reply:
yes, I think so, thanks!